File: AkitaModel/Flagging/ResultRequest.cs
using System;

namespace Skyware.Lis.AkitaModel.Flagging;

/// <summary>
/// Represents DTO for Flagging api call.
/// </summary>
public class ResultRequest
{

    /// <summary>
    /// Date of birth of the patient.
    /// Default value for is 33 years and 6 months old.
    /// </summary>
    public DateTime? DateOfBirth { get; set; } = DateTime.Now.AddYears(-33).AddMonths(-6); // Default age is 33 y and 6 mo

    /// <summary>
    /// Gender indicator of the patient. By default is null (Unknown).
    /// </summary>
    public bool? IsMale { get; set; } // No default gender

    /// <summary>
    /// Id of the patient's species. By default is 1 (Human).
    /// </summary>
    public int? SpeciesId { get; set; } = 1; // Human in LIS iLab

    /// <summary>
    /// References the taken test identifier.
    /// </summary>
    public int TestId { get; set; }

    /// <summary>
    /// Text representation of the test result.
    /// </summary>
    public string TextResult { get; set; }

    /// <summary>
    /// Numeric prefixes of the result.
    /// </summary>
    public string NumericPrefix { get; set; }

    /// <summary>
    /// Numeric result of the test.
    /// </summary>
    public decimal? NumericResult { get; set; }

    /// <summary>
    /// Reference Id, e.g. SaleItemId, Sample barcode, etc.
    /// </summary>
    public string RefId { get; set; }

}

File: AkitaMotelTS/src/Flagging/ResultRequest.ts
/**
 * Skyware.Lis.AkitaModel.Flagging
 */

/**
 * Represents DTO for Flagging API call.
 */
export class ResultRequest {
  /**
   * Date of birth of the patient.
   * Default value for is 33 years and 6 months old.
   */
  dateOfBirth?: Date = new Date(new Date().setFullYear(new Date().getFullYear() - 33, new Date().getMonth() - 6));

  /**
   * Gender indicator of the patient. By default is null (Unknown).
   */
  isMale?: boolean; // No default gender

  /**
   * Id of the patient's species. By default is 1 (Human).
   */
  speciesId?: number = 1; // Human in LIS iLab

  /**
   * Date/time the sample was taken.
   */
  sampleDate?: Date = new Date();

  /**
   * References the taken test identifier.
   */
  testId!: number;

  /**
   * Text representation of the test result.
   */
  textResult?: string;

  /**
   * Numeric prefixes of the result.
   */
  numericPrefix?: string;

  /**
   * Numeric result of the test.
   */
  numericResult?: number;

  /**
   * Reference Id, e.g. SaleItemId, Sample barcode, etc.
   */
  refId?: string;
}
